primitives/src/fixed.rs

```diff
@@ -255,9 +274,57 @@ mod tests {
     }
 
     #[test]
-    #[should_panic]
-    fn deserialize_from_json_should_fail() {
-        serde_json::from_str("-10.00").unwrap()
+    fn deserialize_negative_value_from_json_should_fail() {
+        let serialized = serde_json::to_string(&Fixed::negative_one())
+            .expect("Should be possible to serialize any `Fixed`");
+        let result: Result<Fixed, _> = serde_json::from_str(&serialized);
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "-1.0: negative value not allowed"
+        )
+    }
+
+    #[test]
+    fn decode_negative_value_from_parity_scale_should_fail() {
+        let encoded = Fixed::negative_one().encode();
+        let result: Result<Fixed, _> = Fixed::decode(&mut encoded.as_slice());
+        assert!(result.is_err());
```

Comment:
not possible to check failure reason?

Reply:
Yes, sadly without `chain-error` feature we receive just placeholder [`Codec Error`](https://docs.rs/parity-scale-codec/latest/src/parity_scale_codec/error.rs.html#82)